### ISSUE TITLE
'DirectoryIterator' object has no attribute 'nbatches'

### ISSUE THREAD

**Issue by pantherso48** [NervanaSystems/neon#421]
Hi guys,

I am loading my own data using Keras which created a Directory Iterator that is filled with 2 classes of images. After creating my model layers, cost callback, and optimizer I run fit() on my model:
`from keras.preprocessing.image import ImageDataGenerator

train_datagen = ImageDataGenerator(rescale = 1./255,
                                   shear_range = 0.2,
                                   zoom_range = 0.2,
                                   horizontal_flip = True)

test_datagen = ImageDataGenerator(rescale = 1./255)

training_set = train_datagen.flow_from_directory('/data/datasets/catsanddogs/training_set',
                                                 target_size = (64, 64), #need dimensions expected by CNN
                                                 batch_size = 32,
                                                 class_mode = 'binary')

mlp.fit(training_set, optimizer=optimizer, num_epochs=100, cost=cost,
        callbacks=callbacks)`

This is generating the 'DirectoryIterator' object has no attribute 'nbatches'. From your documentation it looks like I need image features(X) and labels(y) for neon to understand the iterator shape. Which I do not have. Do I need to use Aeon to load the images? Also what is the best way to classify the training images, with a csv file and the image name? Not sure on the implementation.

Thanks!

**Comment by wei-v-wang** [comment]
Hi @mrysztow Could you please take a look?
